src/hazelcore/dynamics/EvaluatorStep2.re

```diff
@@ -0,0 +1,331 @@
+open Sexplib.Std;
+
+[@deriving sexp]
+type step_status =
+  | InvalidInput(int)
+  | BoxedValue(DHExp.t)
+  | Indet(DHExp.t)
+  | Step(DHExp.t);
+
+let rec step_evaluate = (d: DHExp.t): step_status =>
```

Comment:
let's still use an evaluation context like in your other code -- that will be useful in the user interface to be able to highlight (and even allow the user to select) the next steppable expression

Reply:
also note that "evaluate" means "turn into a value", so I wouldn't call a function like this "step_evaluate" -- just "step" is fine.